packages/cdk-assets/lib/private/placeholders.ts

```diff
@@ -0,0 +1,52 @@
+import { AssetManifest } from "../asset-manifest";
+import { IAws } from "../aws-operations";
+
+/**
+ * Replace the {ACCOUNT} and {REGION} placeholders in all strings found in a complex object.
+ */
+export async function replaceAwsPlaceholders<A extends { region?: string }>(object: A, aws: IAws): Promise<A> {
```

Comment:
Feels weird that this is async. I would expect this function signature to be:

```ts
replaceEnvironmentPlaceholders(text: string, replacements: { account: string, region: string }): string;
```

I think it can operate at the string level without dealing with JSON structures...